python/cudf/cudf/core/column/column.py

```diff
@@ -926,16 +926,16 @@ def get_slice_bound(self, label: ScalarLike, side: str, kind: str) -> int:
         else:
             raise ValueError(f"Invalid value for side: {side}")
 
-    def sort_by_values(
+    def sort_values(
         self: ColumnBase,
         ascending: bool = True,
         na_position: str = "last",
-    ) -> Tuple[ColumnBase, "cudf.core.column.NumericalColumn"]:
+    ) -> ColumnBase:
         col_inds = self.as_frame()._get_sorted_inds(
             ascending=ascending, na_position=na_position
         )
-        col_keys = self.take(col_inds)
-        return col_keys, col_inds
+        col_keys = self.take(col_inds, check_bounds=False)
```

Comment:
We should try to _directly_ sort, rather than converting to a frame, getting the sorted indices, and then gathering.